crates/harp/src/routines.rs

```diff
@@ -5,18 +5,32 @@
 //
 //
 
+use std::sync::Mutex;
+
 use libr::R_CallMethodDef;
 use libr::R_getEmbeddingDllInfo;
 use libr::R_registerRoutines;
 use log::error;
 
-static mut R_ROUTINES: Vec<R_CallMethodDef> = vec![];
+// Sync and Send wrapper that we can store in a global. Necessary because
+// `R_CallMethodDef` includes raw pointers.
+struct CallMethodDef {
+    inner: R_CallMethodDef,
+}
+
+unsafe impl Send for CallMethodDef {}
+unsafe impl Sync for CallMethodDef {}
+
+static R_ROUTINES: Mutex<Vec<CallMethodDef>> = Mutex::new(vec![]);
```

Comment:
Nice I also tried this when fiddling with this one last week but got stuck on the `R_CallMethodDef` not being send+sync problem